### PR TITLE
Clean up tests to diagnose Frank-Wolfe linear solver error

### PR DESCRIPTION
**Pull request by gdalle**
- Revert specification of test dependencies to the [more robust version](https://pkgdocs.julialang.org/v1/creating-packages/#Test-specific-dependencies-in-Julia-1.0-and-1.1) (see [this issue](https://github.com/JuliaLang/Pkg.jl/issues/1585))
- Adjust default parameters for Frank-Wolfe (increase precision and switch step size selection) because a recent release or FrankWolfe.jl broke our tests (see #40)
- Deactivate test verbosity to make CI logs easier to read
- Switch data to Float32 to suppress Flux.jl warning
- Try to parallelize tests but they no longer show up in the test set so I gave up (see [this discourse thread](https://discourse.julialang.org/t/run-pkg-test-with-several-processors/6917/10), it seems like an open problem)

### CODE REVIEW

File: test/argmax.jl [hunk 1]
```diff
@@ -1,3 +1,4 @@
+using Base.Threads
```
Comment (BatyLeo) [inline test/argmax.jl:1]: Remove it if not used

File: test/argmax.jl [hunk 2]
```diff
@@ -108,8 +112,8 @@ data_train, data_test = generate_dataset(
 
 ## Test loop
 
-for pipeline in pipelines_imitation_θ
-    pipeline_1 = deepcopy(pipeline)
+for k in eachindex(pipelines_imitation_θ)
```
Comment (BatyLeo) [inline test/argmax.jl:115]: Why do we need the index here ?

File: test/argmax.jl [hunk 3]
```diff
@@ -122,30 +126,30 @@ for pipeline in pipelines_imitation_θ
...
         epochs=100,
-        verbose=true,
+        verbose=verbose,
         setting_name="argmax - imitation_θ - precomputed y_true",
     )
 end
 
-for pipeline in pipelines_imitation_y
-    pipeline = deepcopy(pipeline)
+for k in eachindex(pipelines_imitation_y)
```
Comment (BatyLeo) [inline test/argmax.jl:151]: same question

File: test/argmax.jl [hunk 4]
```diff
@@ -158,13 +162,13 @@ for pipeline in pipelines_imitation_y
...
         epochs=200,
-        verbose=true,
+        verbose=verbose,
         setting_name="argmax - imitation_y",
     )
 end
 
-for pipeline in pipelines_experience
-    pipeline = deepcopy(pipeline)
+for k in eachindex(pipelines_experience)
```
Comment (BatyLeo) [inline test/argmax.jl:170]: same question

File: test/paths.jl [hunk 5]
```diff
@@ -1,3 +1,4 @@
+using Base.Threads
```
Comment (BatyLeo) [inline test/paths.jl:1]: remove if not used

File: test/paths.jl [hunk 6]
```diff
@@ -151,8 +155,8 @@ data_train, data_test = generate_dataset(
 
 ## Test loop
 
-for pipeline in pipelines_imitation_θ
-    pipeline_1 = deepcopy(pipeline)
+for k in eachindex(pipelines_imitation_θ)
```
Comment (BatyLeo) [inline test/paths.jl:158]: same question

File: test/paths.jl [hunk 7]
```diff
@@ -165,30 +169,30 @@ for pipeline in pipelines_imitation_θ
...
         epochs=100,
-        verbose=true,
+        verbose=verbose,
         setting_name="paths - imitation_θ - precomputed y_true",
     )
 end
 
-for pipeline in pipelines_imitation_y
-    pipeline = deepcopy(pipeline)
+for k in eachindex(pipelines_imitation_y)
```
Comment (BatyLeo) [inline test/paths.jl:194]: same question

File: test/paths.jl [hunk 8]
```diff
@@ -201,13 +205,13 @@ for pipeline in pipelines_imitation_y
...
         epochs=200,
-        verbose=true,
+        verbose=verbose,
         setting_name="paths - imitation_y",
     )
 end
 
-for pipeline in pipelines_experience
-    pipeline = deepcopy(pipeline)
+for k in eachindex(pipelines_experience)
```
Comment (BatyLeo) [inline test/paths.jl:213]: same question

File: test/ranking.jl [hunk 9]
```diff
@@ -1,3 +1,4 @@
+using Base.Threads
```
Comment (BatyLeo) [inline test/ranking.jl:1]: remove if not used

File: test/ranking.jl [hunk 10]
```diff
@@ -102,8 +106,8 @@ data_train, data_test = generate_dataset(
 
 ## Test loop
 
-for pipeline in pipelines_imitation_θ
-    pipeline_1 = deepcopy(pipeline)
+for k in eachindex(pipelines_imitation_θ)
```
Comment (BatyLeo) [inline test/ranking.jl:109]: same question

File: test/ranking.jl [hunk 11]
```diff
@@ -116,30 +120,30 @@ for pipeline in pipelines_imitation_θ
...
         epochs=100,
-        verbose=true,
+        verbose=verbose,
         setting_name="ranking - imitation_θ - precomputed_y_true",
     )
 end
 
-for pipeline in pipelines_imitation_y
-    pipeline = deepcopy(pipeline)
+for k in eachindex(pipelines_imitation_y)
```
Comment (BatyLeo) [inline test/ranking.jl:145]: same question

File: test/ranking.jl [hunk 12]
```diff
@@ -152,13 +156,13 @@ for pipeline in pipelines_imitation_y
...
         epochs=200,
-        verbose=true,
+        verbose=verbose,
         setting_name="ranking - imitation_y",
     )
 end
 
-for pipeline in pipelines_experience
-    pipeline = deepcopy(pipeline)
+for k in eachindex(pipelines_experience)
```
Comment (BatyLeo) [inline test/ranking.jl:164]: same question

**Comment by gdalle** [comment]
@BatyLeo maybe switching to agnostic step size is not a good idea, see https://github.com/ZIB-IOL/FrankWolfe.jl/issues/387#issuecomment-1449603456

Let's wait a bit before we merge

File: test/argmax.jl (hunk 2 above)
Comment (gdalle) [reply test/argmax.jl:115]: It was a remainder of my attempt at multithreading tests

Comment (gdalle) [reply test/argmax.jl:115]: https://discourse.julialang.org/t/multi-threaded-testing/95372

**Comment by BatyLeo** [comment]
> @BatyLeo maybe switching to agnostic step size is not a good idea, see [ZIB-IOL/FrankWolfe.jl#387 (comment)](https://github.com/ZIB-IOL/FrankWolfe.jl/issues/387#issuecomment-1449603456)
> 
> Let's wait a bit before we merge

Ok :+1:

**Pull request merged**